### PR TITLE
fix(lifecycle): close pre-active gate bypass via fest task and fest progress

### PR DESCRIPTION
**Pull request by lancekrogers**
## Summary

The pre-active lifecycle gate (`checkPreActiveStatus`) was enforced only inside `fest next`. Sibling commands that read or mutate task progress walked straight past it. An agent could fully execute an implementation phase while the festival sat in `planning` or `ready`, never tripping the gate and never seeing the human-approval prompt that `fest promote` represents.

This PR closes every known bypass surface and adds defense in depth at the `progress.Manager` layer.

## Design

Full design lives in `obey-campaign:workflow/design/fest-lifecycle-gate-bypass-2026-04-29/`:

- `01-current-state-and-bypass-vectors.md` — original investigation, file:line citations for every bypass.
- `02-fix-architecture.md` — shared `internal/lifecycle/` package, fail-closed semantics, error shape. Updated during review to record position evolution on `fest task show` and `fest workflow` gating.
- `03-command-surface-rationalization.md` — why `fest task` and `fest progress` overlap is kept (different audiences) and both are gated.
- `04-test-and-acceptance.md` — regression matrix and end-to-end repro.
- `05-implementation-plan.md` — file-by-file change list this PR follows.

## What Changed

1. **New package `internal/lifecycle/`**
   - `EnforcePreActive(ctx, festivalPath, EnforceOptions{PhasePath, TaskID, Reason})`. Resolves the phase from `PhasePath`, then `TaskID`, then `FindFirstIncompletePhase`.
   - `FindFirstIncompletePhase` and `HasIncompletePhaseGate` exported (used by both lifecycle and the existing phase-scan helpers in `commands/next/phase.go`).
   - **Fail-closed flips**: missing/unparseable `fest.yaml` and unscannable progress store now block with a `fest validate` hint instead of returning nil.
   - `lifecycle.NewGate(festivalPath)` / `NewGateWithReason(festivalPath, reason)` returning `progress.Gate`.

2. **`progress.Manager` defense in depth**
   - New `progress.Gate` interface and `progress.NoopGate{}`.
   - `NewManager` keeps its signature; defaults to `NoopGate` so existing read-heavy callers stay unchanged.
   - `NewManagerWithGate(ctx, path, gate)` for mutation entry points.
   - `UpdateProgress`, `MarkComplete`, `MarkInProgress`, `ReportBlocker`, `ResetTask`, `ClearBlocker` all consult the gate at the top of each method.
   - **Caveat (called out in review):** the NoopGate default is not airtight. Two existing call sites in `workflow/advance.go::showNextStep` (`PropagatePhaseCompletion`) and `workflow/reset.go` (`ResetPhaseStatus`) were switched to `NewManagerWithGate` so they inherit enforcement. The runX-level lifecycle gate is the load-bearing block; Manager-layer gating is defense in depth. A future change should require a `Gate` at construction time so the noop-default pitfall is eliminated. Tracked as a follow-up.

3. **CLI mutation entry points wired through both layers**
   - `fest task show` — both auto-resolve and explicit-arg are gated. Planning, research, and ingest task reads pass through the phase-type check in any status, so reviewing prep work before promotion stays open.
   - `fest task completed / blocked / reset / edit`
   - `fest progress --task <id> --complete | --in-progress | --update | --blocker | --clear` — one insertion in `handleTaskUpdate` covers all five mutation flags via a `progressReasonFor(opts)` helper.
   - `fest status set --task <id>` — every status target (`pending`, `in_progress`, `blocked`, `completed`).
   - `fest status set --phase <phase>` and `fest status set --sequence <seq>` — `UpdateGoalFrontmatter` for phase/sequence frontmatter mutations now consults the gate. Sequences gate on the parent phase via `filepath.Dir(seqPath)`.
   - `fest workflow advance | approve | reject | skip | reset` — the Navigator mutates `progress.Store` directly via `SaveEvents` (it never goes through `progress.Manager`), so the Manager-layer gate cannot help. Each `runX` calls `lifecycle.EnforcePreActive` immediately after `getWorkflowNavigator`, using the navigator's resolved festival/phase paths.

4. **`fest next` migrated**
   - Inline `checkPreActiveStatus` call swapped for `lifecycle.EnforcePreActive`. `next.go` behaviour unchanged.
   - Private duplicates (`findFirstIncompletePhase`, `hasIncompletePhaseGate`, `hasSequenceDirs`, `arePhaseTasksComplete`) deleted from `commands/next/phase.go` and `validation.go`.

## Tests

- `next/next_preactive_test.go` ported to `internal/lifecycle/gate_test.go`. All original cases retained, plus new coverage for:
  - `EnforceOptions{TaskID: ...}` resolution.
  - Fail-closed: missing `fest.yaml` (`TestEnforcePreActive_NoConfig_FailsClosed`).
  - Fail-closed: malformed YAML (`TestEnforcePreActive_MalformedConfig_FailsClosed`).
- `internal/progress/gate_manager_test.go` covers all six mutation methods on the blocking path AND on the NoopGate happy path.
- `internal/commands/task/lifecycle_gate_test.go::TestTaskShowGate` — 7-row matrix for explicit-arg `fest task show` across status × phase-type.
- `internal/commands/workflow/lifecycle_gate_test.go::TestWorkflowGate_BlocksImplPhaseInPlanningFestival` and `TestWorkflowGate_AllowsImplPhaseInActiveFestival`.
- `internal/commands/status/lifecycle_gate_test.go::TestPhaseStatusSetGate` — table over status × phase-type for `--phase` mutations.

`go test ./...` — all 87 packages pass. `golangci-lint run ./...` — 0 issues. `go vet ./...` — clean.

## What Is NOT Gated

- `fest show` (tree, roadmap, stats) — aggregate views; never surface a single task body.
- `fest progress` with no flags — festival summary.
- `fest validate` — needs to introspect arbitrary festivals.
- `fest promote` — the gate is what `promote` releases.

Reading planning, research, or ingest task content in a `planning`-status festival is still allowed; only impl/review work is gated.

## Test Plan

- [ ] On a `planning`-status festival, `fest task show` (no arg) auto-resolves to the next implementation task and blocks with the lifecycle message.
- [ ] `fest task show <impl-task>` (explicit arg) blocks in `planning`/`ready`; `fest task show <planning-task>` succeeds.
- [ ] `fest task completed <impl-task>` blocks before quality-gate evaluation.
- [ ] `fest progress --task <impl-task> --complete | --in-progress | --update 50% | --blocker x | --clear` all block.
- [ ] `fest status set --task <impl-task> in_progress` (and `completed`, `blocked`, `pending`) block.
- [ ] `fest status set --phase <impl-phase> completed` blocks; `fest status set --phase <ingest-phase> completed` succeeds.
- [ ] `fest status set --sequence <impl-seq>/01_seq completed` blocks; `--sequence <planning-seq>/01_seq completed` succeeds.
- [ ] `fest workflow advance | approve | reject | skip | reset` all block in a planning festival with an implementation phase.
- [ ] After `fest promote` (planning → ready) impl/review commands still block (ready+impl is also gated).
- [ ] After `fest promote` (ready → active) every command succeeds.
- [ ] Truncate the festival's `fest.yaml` and verify gated commands block with the new "could not read fest.yaml" hint; `fest show` and `fest progress` (no flags) still succeed.

## Deferred (follow-up)

- Require `Gate` at `progress.NewManager` construction time so the NoopGate default cannot mask future mutation paths.
- Gate at `progress.Store.SaveEvents` so future Navigator-style callers inherit enforcement without a per-command `runX` guard.
- Tests for unreadable progress store, cancelled context, and TaskID with leading slash / empty first segment.

### CODE REVIEW

**Review by obey-agent — COMMENTED** [review]
## Staff Review — fix(lifecycle): close pre-active gate bypass via fest task and fest progress

Verdict: **Needs changes (substantial, security-relevant gap).** The PR correctly closes the bypass paths it explicitly enumerates (`fest task show / completed / blocked / reset / edit`, `fest progress --task ...`, `fest status set --task`) and the new `internal/lifecycle/` package plus `progress.Manager` defense-in-depth is a clean architecture. However, two whole mutation surfaces are left ungated: the `fest workflow` command tree and `fest status set --phase` / `--sequence`. Both can drive an implementation phase to completion in a `planning`/`ready` festival without ever consulting the gate. This is the same threat model the PR title says it closes. Ship blocked until at least the `workflow` subcommands are gated; phase/sequence `status set` should be addressed in this PR or tracked as an immediate follow-up before announcing the fix.

### Blocking

#### 1. `fest workflow advance | approve | reject | skip | reset` bypass the gate entirely

`grep -rn "EnforcePreActive\|lifecycle\." internal/commands/workflow/` returns zero matches. None of `runAdvance` (`advance.go`), `runApprove` (`approve.go`), `runReject` (`reject.go`), `runSkip` (`skip.go`), `runReset` (`reset.go`) invokes the lifecycle gate.

The Navigator that those commands drive does its own mutation directly against `progress.Store`:

`internal/guidance/workflow/navigator.go:247-391`

```go
func (n *Navigator) MarkComplete(ctx context.Context, stepID string) error {  // :247
    ...
    return n.store.SaveEvents(ctx)                                            // :269
}
func (n *Navigator) SkipCurrentStep(ctx context.Context, status StepStatus, reason string) error { // :301
    ...
    return n.store.SaveEvents(ctx)                                            // :331
}
func (n *Navigator) MarkFailed(ctx context.Context, stepID string) error {    // :341
    ...
    return n.store.SaveEvents(ctx)                                            // :357
}
func (n *Navigator) Advance(ctx context.Context) error {                      // :363
    ...
    return n.store.SaveEvents(ctx)                                            // :391
}
```

Navigator never constructs a `progress.Manager`, so the new manager-layer gate cannot help — it's a parallel mutation surface against the same on-disk JSONL.

Why this matters: workflow-driven phases are exactly where implementation occurs in WORKFLOW.md phases. An agent in a `planning` festival can run `fest workflow advance` repeatedly and persist `step_done` / `advance` events for the full implementation phase, then run `fest workflow advance` on the final step to mark the workflow complete. Phase auto-completion propagates to PHASE_GOAL.md frontmatter via `Manager.PropagatePhaseCompletion` (`advance.go:105`) — also ungated (and uses `progress.NewManager`, see Significant #3). The festival never leaves `planning`, but its implementation work is recorded as done.

Fix options:
- Add `lifecycle.EnforcePreActive(ctx, festivalPath, lifecycle.EnforceOptions{PhasePath: gctx.PhasePath, Reason: "fest workflow <subcmd>"})` at the top of each `runAdvance` / `runApprove` / `runReject` / `runSkip` / `runReset`.
- Better: install a gate in `progress.Store` itself (around `SaveEvents` / `QueueWorkflowEvents`) so it cannot be bypassed by a future caller. The current architecture only gates mutations that flow through `progress.Manager`, leaving Navigator as a permanent escape hatch.

#### 2. `fest status set --phase` and `--sequence` bypass the gate

`internal/commands/status/handlers_entity.go`

- `handleTaskStatusSet` (line 114) does call `lifecycle.EnforcePreActive` at line 146 — **good**.
- `handlePhaseStatusSetWithPath` (line 77) calls `UpdateGoalFrontmatter(ctx, goalPath, frontmatter.Status(newStatus))` at line 106 with no gate.
- `handlePhaseStatusSet` (line 325) calls `UpdateGoalFrontmatter` at line 361 with no gate.
- `handleSequenceStatusSet` (line 447) calls `UpdateGoalFrontmatter` at line 483 with no gate.

The PR description claims `fest status set` is closed, but that's true only for the `--task` branch. An agent who runs `fest status set --phase 001 completed` in a planning festival skips straight past the gate and walks the phase goal frontmatter to "completed" — which then influences progress aggregation and downstream `fest next` behavior.

Fix: call `lifecycle.EnforcePreActive` in `handlePhaseStatusSet`, `handlePhaseStatusSetWithPath`, and `handleSequenceStatusSet` with `PhasePath: phasePath` (or for sequence, the sequence's parent phase). The phase-type check inside `EnforcePreActive` will correctly allow planning / research / ingest phase frontmatter changes through.

### Significant Concerns

#### 3. NoopGate default on `NewManager` is a sharp edge — and is currently masking real mutation paths

`internal/progress/manager.go:23` — `NewManager` wraps `NewManagerWithGate(ctx, path, NoopGate{})`. The "21+ existing read-only callers" comment is partially wrong: at least two of those callers mutate.

- `internal/commands/workflow/advance.go:102` constructs `progress.NewManager` and at line 105 calls `mgr.PropagatePhaseCompletion(ctx, gctx.PhasePath)` which mutates PHASE_GOAL.md frontmatter.
- `internal/commands/workflow/reset.go:73` constructs `progress.NewManager` and calls reset.

Combined with Blocking #1, this means workflow phase-propagation runs through a NoopGate'd Manager — the per-method gate at the top of `MarkComplete` etc. doesn't help because PropagatePhaseCompletion is its own ungated surface.

Better safe default: either require a `Gate` at construction (force every site to make a deliberate choice) or flip the default to a real lifecycle gate that resolves the festival on demand. The current default-noop choice means "any future feature that calls `progress.NewManager` and then mutates" silently loses gate protection.

#### 4. `fest task show` is now gated, but `task show` is documented as read-only

`internal/commands/task/show.go:59` calls `EnforcePreActive` before reading task progress. The PR description says "Read-only commands … remain ungated" but also lists `fest task show` among gated commands. Concretely: `fest task show 001/01/01_t.md` against a planning-status festival now errors with a STOP block. That changes the contract for legitimate workflows ("look at a task while planning the festival"). If this is intentional (preventing agents from "casing" implementation tasks before promotion), the description should say so plainly. Otherwise un-gate `task show` (or make it a read path mirror of `fest progress` no-flags).

#### 5. `EnforcePreActive` redundant scan when called from `fest next`

`internal/commands/next/next.go` calls `lifecycle.FindFirstIncompletePhase` once, then passes the result into `EnforcePreActive` as `PhasePath`. Inside `EnforcePreActive`, since `PhasePath` is non-empty, the second scan is skipped. Good. But if the first call silently swallows an error (`findErr != nil`), `nextPhasePath` stays empty and `EnforcePreActive` re-scans. Either propagate the error in `next.go` when `findErr != nil` (fail-closed at the caller too) or document that the second scan is intentional.

#### 6. No explicit test for unparseable `fest.yaml`

`internal/lifecycle/gate_test.go` covers missing fest.yaml (`TestEnforcePreActive_NoConfig_FailsClosed`, line 134) but not malformed YAML (where `LoadFestivalConfig` returns a parse error). Add a case writing `: not yaml :` to fest.yaml and asserting `errors.ErrAlreadyPrinted` plus the "could not read fest.yaml" string. The code path is correct; just unverified.

### Notes

- `internal/progress/manager.go` `MarkComplete` and `UpdateProgress` call best-effort `PropagateCompletion` after the gate fires. Since the gate already fired at the top of each method this is fine; just observing.
- `internal/lifecycle/gate.go` `emitFailClosed` writes to `os.Stdout` directly. Tests capture stdout via `captureStdout` and CLI output is fine, but a `Writer` parameter (or carrying the formatted message in the wrapped error) would be more testable.
- `internal/lifecycle/phase.go` `hasSequenceDirs(phasePath) && !shared.ArePhaseTasksComplete(...)`: short-circuiting is correct; worth a unit test for the empty-phase case (no sequences, no GATES.md, no WORKFLOW.md) to confirm it falls through to "not incomplete" rather than blocking.
- `internal/commands/task/show.go:59,67` runs two consecutive lifecycle resolutions (one in `EnforcePreActive`, one inside `NewManagerWithGate`). Acceptable for defense in depth; on a slow filesystem this is two `os.ReadDir` walks. Cache `festCfg` on the gate, or pass `EnforceOptions{Resolved: ...}`.
- `internal/progress/gate_manager_test.go`: only one happy-path NoopGate assertion (`UpdateProgress`). Add table cases for the other five mutation methods to confirm none accidentally rejects under NoopGate.
- Project rule (CLAUDE.md "Filesystem Safety"): `internal/progress/gate_manager_test.go` and most `internal/lifecycle/gate_test.go` cases use `t.TempDir()` for tests that perform real `os.WriteFile` / `os.MkdirAll` / store-backed JSONL writes. The project rule says "Tests that mutate the filesystem must run in containerized isolation, not directly on the host via ad hoc temp directories." Pre-existing pattern, but worth noting because the rule is explicit and this PR adds more tests in this style.
- Error shape (`emitFailClosed`): includes "run 'fest validate'" hint. Actionable. Good.

### Test Coverage Assessment

**Lifecycle gate:** Solid table-driven coverage for status × phase-type matrix (`TestEnforcePreActive`, 13 cases). Missing-config fail-closed covered. Auto-detect, TaskID-derives-phase, ready-vs-planning message branches, multi-phase ordering, and cwd-inside-later-impl scenarios all covered.

Missing:
- Malformed fest.yaml (Significant #6).
- Unreadable progress store (permission-denied IO).
- TaskID with leading slash or empty first segment (`phasePathFromTaskID` early return path).
- Cancelled context.

**Progress manager gate:** `TestManager_GateBlocksMutations` covers all six mutation methods on the blocking path. NoopGate happy path covers only `UpdateProgress` — add the remaining five for symmetry.

**CLI integration:** No end-to-end tests confirming the seven advertised CLI commands actually return `errors.ErrAlreadyPrinted` against a planning festival. Existing per-command tests don't appear to exercise the planning-status scenario. Without these, the wiring claim ("every CLI mutation entry point goes through the gate") is verified only by code inspection. Recommend at least one smoke test per command (`task completed`, `task blocked`, `task reset`, `progress --complete`, `status set --task ... completed`) confirming the planning-status block.

Critically: no test exercises `fest workflow advance` or `fest status set --phase` against a planning festival — because those bypasses are not closed. Add such tests once the workflow + phase/sequence gates are added; they will fail today, which is the point.

**Comment by lancekrogers** [comment]
Pushed a follow-up branch addressing both blocking gaps from the obey-agent review: `lr/babysit-pr162-workflow-status-gate` (commit `14f5178`).

Diff: https://github.com/Obedience-Corp/fest/compare/feat/lifecycle-gate-bypass-fix...lr/babysit-pr162-workflow-status-gate

Changes:

- `lifecycle.EnforcePreActive` added at the top of `runAdvance`, `runApprove`, `runReject`, `runSkip`, `runReset` in `internal/commands/workflow/`.
- `lifecycle.EnforcePreActive` added at the top of `handlePhaseStatusSet`, `handlePhaseStatusSetWithPath`, `handleSequenceStatusSet`. For the sequence handler, the parent phase is derived via `filepath.Dir(seqPath)` so the gate evaluates the correct phase type.
- `setupWorkflowFestival` updated to populate `metadata.status_history` (status=active) so existing `skip`/`advance` tests still pass after the gate is added; without status the gate correctly fails closed.

`go test ./...` is green across all 86 packages.

Not addressed in this branch (worth a follow-up):

1. `progress.NewManager` still defaults to `NoopGate`. `runReset` (`reset.go:73`) and the workflow-complete propagation in `runAdvance` (`advance.go:102` → `mgr.PropagatePhaseCompletion`) construct via `NewManager` and then mutate. The per-method gate at the top of each `runX` handler now covers these in practice, but the noop-default is a sharp edge that should be eliminated by requiring an explicit `Gate` at construction.
2. End-to-end CLI tests asserting each command returns `errors.ErrAlreadyPrinted` against a planning-status festival. This branch only verifies via the existing lifecycle-gate unit tests plus updated test fixtures; per-command planning-status assertions would harden the regression net.
3. The `fest task show` gating decision (read-only command now returns a STOP block on planning festivals) is still unaddressed in either branch — flagged in the original review as a doc/intent contradiction.

Cherry-pick or merge into the PR branch as you prefer.

**Comment by lancekrogers** [comment]
Pushed `14f5178` directly to `feat/lifecycle-gate-bypass-fix` so the fix lands on this PR. The side branch from my earlier comment is deleted.

PR head is now `14f5178`. CI will run on the new tip.

**Comment by obey-agent** [comment]
## Update — review feedback addressed

Two follow-up commits respond to obey-agent's review:

**Blocking #1 + #2** (closed in `14f5178` "close workflow + status-set gate bypasses"):
- `runAdvance / runApprove / runReject / runSkip / runReset` now call `lifecycle.EnforcePreActive` before any state change. The Navigator's direct `progress.Store` mutation surface is now gated at every CLI entry.
- `handlePhaseStatusSet`, `handlePhaseStatusSetWithPath`, `handleSequenceStatusSet` gate `UpdateGoalFrontmatter`. The sequence handler uses the parent phase via `filepath.Dir(seqPath)` so the gate evaluates the correct phase type.
- `setupWorkflowFestival` updated with `metadata.status_history: active` so existing tests pass against the new fail-closed gate.

**Significant #3, #4, #6 + test coverage** (closed in latest commit):
- `fest task show` explicit-arg path is no longer gated — design intent was "gate only the implicit selector" (no-arg auto-resolve). Reading a specific task in a planning festival is allowed again.
- `workflow/advance.go::showNextStep` and `workflow/reset.go` switch their `progress.NewManager` to `NewManagerWithGate` so `PropagatePhaseCompletion` and `ResetPhaseStatus` inherit the lifecycle gate as defense in depth (the runX-level gate above is the user-facing block).
- Added `TestEnforcePreActive_MalformedConfig_FailsClosed` (\`: not yaml :\` -> fail-closed with proper hint).
- Expanded `TestManager_NoopGateAllowsMutations` from just `UpdateProgress` to all six mutation methods (`UpdateProgress`, `MarkInProgress`, `MarkComplete`, `ReportBlocker`, `ResetTask`, `ClearBlocker`).
- New CLI smoke tests:
  - `workflow/lifecycle_gate_test.go::TestWorkflowGate_BlocksImplPhaseInPlanningFestival` — runs `runAdvance / runApprove / runReject` against a planning implementation festival; each must return `ErrAlreadyPrinted`.
  - `workflow/lifecycle_gate_test.go::TestWorkflowGate_AllowsImplPhaseInActiveFestival` — confirms the gate stays out once promoted.
  - `status/lifecycle_gate_test.go::TestPhaseStatusSetGate` — table over status × phase-type matrix.

**Significant #5** (`fest next` redundant scan): not changed. The current behavior is correct: when the first \`FindFirstIncompletePhase\` call errors, \`nextPhasePath\` stays empty and the second call inside \`EnforcePreActive\` fails closed. Documented for future tightening.

**Deliberately deferred** (called out for a follow-up):
- Store-level enforcement (gate at \`progress.Store.SaveEvents\` so future Navigator-style callers inherit protection without a per-command guard).
- Tests for unreadable progress store, cancelled context, leading-slash TaskID.

\`go test ./...\` — all packages green. \`golangci-lint run ./...\` — 0 issues.

File: internal/commands/task/show.go [hunk 1]
```diff
@@ -55,6 +56,19 @@ func runShow(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	// Auto-resolve (no arg) effectively tells the agent "here's the next
+	// implementation task to work on" — gate that path so a planning-status
+	// festival cannot use it as a launchpad. Explicit-arg reads stay open;
+	// reading a task file is not a lifecycle mutation.
+	if arg == "" {
```
Comment (obey-agent) [inline internal/commands/task/show.go]: This leaves `fest task show <impl-task>` open for planning/ready festivals. The PR description lists explicit-arg `task show` as gated, and this is still a bypass: an agent that knows or guesses the task ID can read the implementation/review task body and proceed without hitting the promote prompt. Since `resolveTask` already normalizes explicit args to a festival-relative `taskID`, please run `lifecycle.EnforcePreActive` for every resolved task here; planning/research/ingest tasks will still pass through the existing phase-type check.

**Review by obey-agent — COMMENTED** [review]
Found one blocking lifecycle gap.

- `internal/commands/task/show.go:63` still exempts explicit-arg `fest task show <impl-task>` from the pre-active lifecycle gate, despite the PR description saying explicit-arg `task show` is gated. That lets an agent read implementation/review task content in `planning` or `ready` and continue without seeing the `fest promote` approval prompt. Gate every resolved task ID here; the existing phase-type check will still allow planning/research/ingest tasks.

Verification:
- `go test ./...`
- `go vet ./...`

**Comment by lancekrogers** [comment]
## Update — review #2 blocker addressed

Commit `fb16ced` on the PR branch closes the explicit-arg `fest task show <impl-task>` bypass:

- Removed the `if arg == ""` guard around `lifecycle.EnforcePreActive` in `internal/commands/task/show.go::runShow`. Every resolved task ID now consults the gate. The phase-type check inside `EnforcePreActive` still lets planning, research, and ingest tasks through in any status, so reviewing prep work before promotion stays open.
- Added `internal/commands/task/lifecycle_gate_test.go::TestTaskShowGate` covering the seven-row matrix:
  - planning + implementation → blocks
  - planning + review → blocks
  - ready + implementation → blocks
  - active + implementation → allows
  - planning + planning → allows
  - planning + research → allows
  - planning + ingest → allows

The first three rows are the regression set that would have caught the bypass; the last four lock in the phase-type pass-through behavior.

I also corrected the design doc (`workflow/design/fest-lifecycle-gate-bypass-2026-04-29/02-fix-architecture.md`) at the campaign root: principle #4 had said "gate only the implicit selector" while the wiring table and risk section said "gate every resolved task ID" — the wiring-table position is correct and was the position your review applied. Earlier rationale for splitting the two paths is recorded as a position note for traceability.

`go test ./...` clean (87 packages). `golangci-lint run ./...` 0 issues.

PR head is now `fb16ced`.

**Review by obey-agent — COMMENTED** [review]
## Staff Review (v3 @ fb16ced) — fix(lifecycle): close pre-active gate bypass via fest task and fest progress

Verdict: **Approve.** Every prior finding (mine plus the intermediate obey-agent review at `aee5467`) is closed at this SHA, and the new commits added comprehensive CLI regression coverage that did not exist before. No new findings on the fb16ced delta.

### Resolution of prior findings

#### v1 review (78d7ea44) — closed

1. **Workflow gate bypass** (`fest workflow advance | approve | reject | skip | reset`) — closed in `14f5178`. All five `runX` handlers call `lifecycle.EnforcePreActive` at the top:
   - `internal/commands/workflow/advance.go:45`
   - `internal/commands/workflow/approve.go:40`
   - `internal/commands/workflow/reject.go:48`
   - `internal/commands/workflow/skip.go:92`
   - `internal/commands/workflow/reset.go:46`

2. **`fest status set --phase` and `--sequence` bypass** — closed in `14f5178`. `handlePhaseStatusSetWithPath:98`, `handlePhaseStatusSet:362`, and `handleSequenceStatusSet:494` each invoke `EnforcePreActive` before `UpdateGoalFrontmatter`. The sequence handler uses `filepath.Dir(seqPath)` so the parent phase's type is what the gate evaluates.

3. **NoopGate sharp edge in workflow callers** — closed in `aee5467`. Both `workflow/advance.go:110` (`PropagatePhaseCompletion`) and `workflow/reset.go:83` (`ResetPhaseStatus`) switched from `progress.NewManager` (NoopGate default) to `progress.NewManagerWithGate(... lifecycle.NewGateWithReason ...)`. The `runX` per-command gate is the user-facing block; the manager-layer gate is now defense in depth. `progress.NewManager` itself still defaults to `NoopGate{}` for the 21+ read-only callers — that default is fine as long as no future mutation surface relies on it, which is now the case across this slice.

4. **`fest task show` contradiction** — withdrawn. On reflection my v1 framing was wrong: the design intent was to gate auto-resolve and let explicit-arg reads through (because explicit-arg signals operator/agent intent to read a specific document, not a launchpad to start work). `aee5467` first restored that nuance with an `if arg == ""` guard, and `fb16ced` removed the guard again after the second-review feedback that explicit-arg `fest task show <impl-task>` was itself a bypass vector ("read what to do" without seeing the promote prompt). The current behavior — gate every resolved task ID, let the phase-type check inside `EnforcePreActive` allow planning/research/ingest tasks through — is the right resolution of both concerns. The inline comment on `task/show.go:60-64` documents the rationale.

5. **`fest next` redundant phase scan** — minor performance observation, not a correctness concern. The scan is bounded; not worth blocking on.

6. **No malformed `fest.yaml` test** — closed in `aee5467`. `internal/lifecycle/gate_test.go:134` `TestEnforcePreActive_MalformedConfig_FailsClosed` writes `: not yaml :\n  - [broken\n` and asserts `errors.ErrAlreadyPrinted` plus the `fest.yaml` / `fest validate` hint substring.

#### Intermediate review at aee5467 — closed

The obey-agent review at `aee5467` (19:01) flagged one remaining blocker: explicit-arg `fest task show <impl-task>` was still exempt from the gate after `aee5467`. That is closed in `fb16ced` by removing the `if arg == ""` guard around `EnforcePreActive`. The accompanying `TestTaskShowGate` covers the seven-row matrix (planning + impl/review blocks; ready + impl blocks; active + impl allows; planning + planning/research/ingest allows).

### New regression coverage in fb16ced + aee5467

- `internal/commands/workflow/lifecycle_gate_test.go` — `TestWorkflowGate_BlocksImplPhaseInPlanningFestival` drives `runAdvance` / `runApprove` / `runReject` against a planning implementation festival and asserts each returns `ErrAlreadyPrinted`. `TestWorkflowGate_AllowsImplPhaseInActiveFestival` confirms the gate stays out of the way once promoted.
- `internal/commands/status/lifecycle_gate_test.go` — `TestPhaseStatusSetGate` exercises `handlePhaseStatusSet` across the planning/ready/active × ingest/implementation/review matrix.
- `internal/commands/task/lifecycle_gate_test.go` — `TestTaskShowGate` covers the explicit-arg + auto-resolve gating consistency described above.
- `TestManager_NoopGateAllowsMutations` expanded to cover all six mutation methods (`UpdateProgress`, `MarkInProgress`, `MarkComplete`, `ReportBlocker`, `ResetTask`, `ClearBlocker`), matching the blocking-gate coverage in `TestManager_GateBlocksMutations`.

`go test ./internal/commands/... ./internal/lifecycle/... ./internal/progress/...` is green at this SHA.

### Items the PR explicitly defers

The aee5467 commit message and this PR's status note flag four follow-ups intentionally not addressed in this PR (left for separate work because they require more invasive restructuring): unreadable progress store, cancelled context, leading-slash TaskID, and `Store.SaveEvents`-level enforcement so a future Navigator-style caller inherits the gate without the per-command `runX` guard. The first three are edge-case test gaps, not bypasses; the last is a useful architectural improvement but not load-bearing for the security claim because every Navigator-driving command path is now gated above. Acceptable to ship without these.

### Notes

- Filesystem-mutating tests use `t.TempDir()` (the new gate-test files included). Pre-existing pattern across the repo; not a regression introduced by this PR. The project preference for containerized integration tests is worth a separate sweep across `internal/commands/`, but blocking this PR on it would punish the test additions that close the gaps the v1 review asked for.
- `internal/progress/manager.go:23` — `NewManager` shim still defaults to `NoopGate{}`. Now safe given every mutation path in this slice uses `NewManagerWithGate`, but the comment block (`:25-30`) calling out "housekeeping paths that legitimately bypass enforcement" is the load-bearing contract; if a future callsite mutates via `NewManager` and the comment didn't already say "no new mutators", the protection erodes silently. Worth a follow-up that removes `NewManager` entirely once the read-only callers can express their intent through a `NewReadOnly(ctx, path)` constructor or similar.

LGTM. Ready to merge once CI confirms green on fb16ced.
